scripts/update_old_checkpoints.py

```diff
@@ -0,0 +1,156 @@
+        config[new_key] = config.pop(old_key)
+    else:
+        logging.warning(f"Key '{old_key}' not found in configuration.")
+
+
+def add_new_keys(config: config_type):
+    model_config = config["model_raw" if "model_raw" in config else "model"]["config"]
+    model_config["use_weight_tying"] = False
+    model_config["use_meta_device"] = False
+
+
+def remove_keys(config: config_type):
+    if "evaluation_subscriber" in config and "experiment_id" in config["evaluation_subscriber"]["config"]:
+        del config["evaluation_subscriber"]["config"]["experiment_id"]
+    if "settings" in config and "experiment_id" in config["settings"]:
+        del config["settings"]["experiment_id"]
+    if (
+        "checkpoint_saving" in config
+        and "checkpoint_saving_execution" in config["checkpoint_saving"]["config"]
+        and "experiment_id" in config["checkpoint_saving"]["config"]["checkpoint_saving_execution"]["config"]
+    ):
+        del config["checkpoint_saving"]["config"]["checkpoint_saving_execution"]["config"]["experiment_id"]
+
+
+def update_model_state_dict(old_model_path: str, new_model_path: str):
+    """
+    Update the model state dictionary by loading the old model and saving it to the new path.
+
+    Args:
+        old_model_path (str): Path to the old model file.
+        new_model_path (str): Path to the new model file.
+    """
+    state_dict = torch.load(old_model_path)
+    if "lm_head.weight" in state_dict:
+        state_dict["transformer.lm_head.weight"] = state_dict["lm_head.weight"]
+        del state_dict["lm_head.weight"]
+        torch.save(state_dict, new_model_path)
+    else:
+        logging.error("'lm_head.weight' not found in the model state dictionary.")
+        if "transformer.lm_head.weight" in state_dict:
+            logging.error("The model state dictionary already seems to be in the updated format.")
+
+
+def test_loading_config(new_config_path: str):
+    with temporary_environ({"LOCAL_RANK": "0", "RANK": "0"}):
+        load_app_config_dict(Path(new_config_path))
+
+
+@contextmanager
+def temporary_environ(env_vars: dict[str, str]):
+    old_env = {}
+    for key, value in env_vars.items():
+        old_env[key] = os.environ.get(key)
+        os.environ[key] = value
+    try:
+        yield
+    finally:
+        for key, value in old_env.items():
+            if value is None:
+                del os.environ[key]
+            else:
+                os.environ[key] = value
+
+
+if __name__ == "__main__":
+    import sys
+
+    if len(sys.argv) < 3:
+        print("Usage: python update_old_checkpoints.py <old_model_config> <new_model_config> [new_checkpoint_path]")
+        print("If only a config file conversion is needed, omit the third argument.")
+        exit(1)
+
+    old_model_config = sys.argv[1]
+    new_model_config = sys.argv[2]
+    new_checkpoint_path = sys.argv[3] if len(sys.argv) > 3 else None
+
+    update_model(old_model_config, new_model_config, new_checkpoint_path)
```

Comment:
I would make updating checkpoint and updating the config two separate functions that get called sequentially here.